[WIP] Improve Snaps concepts and API reference (#1098)
MetaMask/metamask-docs: snaps/concepts/execution-environment.md

Snaps are untrusted JavaScript programs that execute safely in a sandboxed environment that runs
[Secure ECMAScript (SES)](#secure-ecmascript-ses).
This allows MetaMask to restrict access to global JavaScript APIs and to isolate a Snap's code from
other parts of the application.

> There's no DOM, no Node.js built-ins, and no platform-specific APIs other than the default `snap`

Reviewer: ```suggestion
This environment excludes a DOM, Node.js built-ins, and platform-specific APIs, except for the default `snap` 
```
